homeassistant/components/tplink/__init__.py

```diff
@@ -44,6 +48,10 @@
                     cv.ensure_list, [TPLINK_HOST_SCHEMA]
                 ),
                 vol.Optional(CONF_DISCOVERY, default=True): cv.boolean,
+                vol.Optional(CONF_RETRY_DELAY, default=SLEEP_TIME): cv.positive_int,
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow with options flow.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Sorry for the delay, should I close the PR and just open a new one? When refactoring, do I remove the existing option to specify the config via yaml or just add the option to define via the config flow?

Reply:
Never mind on the 2nd question...just read... ```In rare cases, we can make an exception. Existing integrations that should not have a YAML configuration are allowed and encouraged to implement a configuration flow and remove YAML support. ```

I am going to start working on this.

Reply:
If possible we want to use the yaml config options to update the config entry options to make the transition as smooth as possible. After some time, the yaml config can then be removed.